crates/turborepo-lib/src/hash/mod.rs

```diff
@@ -0,0 +1,481 @@
+                EnvMode::Strict => task_hashable::EnvMode::Strict,
+            }
+        }
+    }
+}
+
+struct TaskHashable {
+    // hashes
+    global_hash: String,
+    task_dependency_hashes: Vec<String>,
+    hash_of_files: String,
+    external_deps_hash: String,
+
+    // task
+    package_dir: turbopath::RelativeUnixPathBuf,
+    task: String,
+    outputs: TaskOutputs,
+    pass_thru_args: Vec<String>,
+
+    // env
+    env: Vec<String>,
+    resolved_env_vars: EnvVarPairs,
+    pass_thru_env: Vec<String>,
+    env_mode: EnvMode,
+    dot_env: Vec<turbopath::RelativeUnixPathBuf>,
+}
+
+pub struct GlobalHashable {
+    global_cache_key: String,
+    global_file_hash_map: HashMap<turbopath::RelativeUnixPathBuf, String>,
+    root_external_deps_hash: String,
+    env: Vec<String>,
+    resolved_env_vars: Vec<String>,
+    pass_through_env: Vec<String>,
+    env_mode: EnvMode,
+    framework_inference: bool,
+    dot_env: Vec<turbopath::RelativeUnixPathBuf>,
+}
+
+struct TaskOutputs {
+    inclusions: Vec<String>,
+    exclusions: Vec<String>,
+}
+
+struct LockFilePackages(pub Vec<turborepo_lockfiles::Package>);
+
+struct FileHashes(pub HashMap<turbopath::RelativeUnixPathBuf, String>);
+
+impl From<TaskOutputs> for Builder<HeapAllocator> {
+    fn from(value: TaskOutputs) -> Self {
+        let mut message = ::capnp::message::TypedBuilder::<
+            proto_capnp::task_outputs::Owned,
+            HeapAllocator,
+        >::new_default();
+        let mut builder = message.init_root();
+
+        {
+            let mut inclusions = builder
+                .reborrow()
+                .init_inclusions(value.inclusions.len() as u32);
+            for (i, inclusion) in value.inclusions.iter().enumerate() {
+                inclusions.set(i as u32, inclusion);
+            }
+        }
+
+        {
+            let mut exclusions = builder
+                .reborrow()
+                .init_exclusions(value.exclusions.len() as u32);
+            for (i, exclusion) in value.exclusions.iter().enumerate() {
+                exclusions.set(i as u32, exclusion);
+            }
+        }
+
+        message.into_inner()
+    }
+}
+
+impl From<LockFilePackages> for Builder<HeapAllocator> {
+    fn from(LockFilePackages(packages): LockFilePackages) -> Self {
+        let mut message = ::capnp::message::TypedBuilder::<
+            proto_capnp::lock_file_packages::Owned,
+            HeapAllocator,
+        >::new_default();
+        let mut builder = message.init_root();
+
+        {
+            let mut packages_builder = builder.reborrow().init_packages(packages.len() as u32);
+            for (i, turborepo_lockfiles::Package { key, version }) in packages.iter().enumerate() {
+                let mut package = packages_builder.reborrow().get(i as u32);
+                package.set_key(key);
+                package.set_version(version);
+                // we don't track this in rust, set it to false
+                package.set_found(false);
```

Comment:
I'm assuming we set this to false on the Go side as well before hashing, but when we pass the lockfile transitive closure from Rust to Go we [set found to true for everything](https://github.com/vercel/turbo/blob/main/crates/turborepo-ffi/src/lockfile.rs#L19)

Reply:
Yeah in the [global hash PR](https://github.com/vercel/turbo/pull/5568/files#diff-0d8990c569c001c7e8cf9439823abd0e0893cd383612e213c4f910d39afa1566) we set it to true